### PR TITLE
[Merge] adds execution layer datastructures

### PR DESCRIPTION
**Pull request by tbenr**
## PR Description

adds execution layer datastructures
* `ExecutionPayload`
* `ExecutionPayloadHeader`
* `Transaction`

related to #4504

## Documentation

- [X] I thought about documentation and added the `documentation` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/ExecutionPayload.java [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+      Bytes logs_bloom,
+      Bytes32 random,
+      UInt64 blockNumber,
+      UInt64 gas_limit,
+      UInt64 gas_used,
+      UInt64 timestamp,
+      Bytes extra_data,
+      UInt256 baseFeePerGas,
+      Bytes32 block_hash,
+      List<Bytes> transactions) {
```
Comment (ajsutton) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/ExecutionPayload.java]: nit: Should use camel case for these param names.

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/ExecutionPayload.java [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+            .map(Transaction.SSZ_SCHEMA::createOpaque)
+            .collect(SSZ_SCHEMA.getTransactionsSchema().collector()));
+  }
+
+  @Override
+  public ExecutionPayloadSchema getSchema() {
+    return SSZ_SCHEMA;
+  }
+
+  public Bytes32 getParent_hash() {
```
Comment (ajsutton) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/ExecutionPayload.java]: nit: And camel case for all the getters.

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/ExecutionPayloadHeader.java [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+      Bytes logs_bloom,
+      Bytes32 random,
+      UInt64 blockNumber,
+      UInt64 gas_limit,
+      UInt64 gas_used,
+      UInt64 timestamp,
+      Bytes extra_data,
+      UInt256 baseFeePerGas,
+      Bytes32 block_hash,
+      Bytes32 transactions_root) {
```
Comment (ajsutton) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/ExecutionPayloadHeader.java]: Camel case through here too. :)

**Review by ajsutton — APPROVED** [review]
LGTM - just need to switch from snake_case to camelCase since that's ultimately what we want everywhere.

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/ExecutionPayloadHeader.java (hunk 3 above)
Comment (tbenr) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/ExecutionPayloadHeader.java]: yeah, going to clean this mess :)

**Pull request merged**